instrumentation/http-url-connection/javaagent/src/test/groovy/HttpUrlConnectionTest.groovy

```diff
@@ -258,4 +258,69 @@ class HttpUrlConnectionTest extends HttpClientTest<HttpURLConnection> implements
       }
     }
   }
+
+  def "sun.net.www.protocol.http.HttpURLConnection.getOutputStream should transform GET into POST"() {
+    setup:
+    def url = resolveAddress("/success").toURL()
+    runWithSpan("someTrace") {
+
+      HttpURLConnection connection = url.openConnection()
+
+      def connectionClassName = connection.getClass().getCanonicalName()
+
+      assert "sun.net.www.protocol.http.HttpURLConnection".equals(connectionClassName)
+
+      connection.setRequestMethod("GET")
+
+      String urlParameters = "q=ASDF&w=&e=&r=12345&t="
+
+      // Send GET request
+      connection.setDoOutput(true)
+      DataOutputStream wr = new DataOutputStream(connection.getOutputStream())
```

Comment:
I believe you currently rely on `getOutputStream` being the method first method that is called (where the span is started) adding `connection.connect()` before it seems to break the test.

Reply:
Indeed, I can reproduce.

I disabled the new implementation code and added `connection.connect()` to the test (before `getOutputStream`). It is also failing with the following exception:
```
java.lang.IllegalStateException: Already connected
	at java.base/java.net.URLConnection.checkConnected(URLConnection.java:1866)
	at java.base/java.net.URLConnection.setDoOutput(URLConnection.java:912)
	at HttpUrlConnectionTest.sun.net.www.protocol.http.HttpURLConnection.getOutputStream should transform GET into POST_closure8(HttpUrlConnectionTest.groovy:294)
```

When I add `connection.connect()` to [another test](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/http-url-connection/javaagent/src/test/groovy/HttpUrlConnectionTest.groovy#L210) and disable the new code, the same kind of exception happens.

So, the exception does not seem related to the new code, and the connection appears to be already open somewhere in the test code.

I may miss something about a specific use case with `getOutputStream()` and `connect()` methods. If this is the case, please tell me.

Reply:
Try adding ` connection.connect()` after `connection.setDoOutput(true)`. This way test should fail because it gets `HTTP GET` instead of `HTTP POST` as span name and not with `java.lang.IllegalStateException`.

Reply:
Fixed with the new implementation. Thank you for having pointed this @laurit.